server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1577,7 +1577,7 @@ protected DataStoreMetadataUpdateResult updateDataSourceMetadataWithHandle(
           oldCommitMetadataFromDb,
           startMetadata
       );
-      return DataStoreMetadataUpdateResult.FAILURE;
+      return DataStoreMetadataUpdateResult.TRY_AGAIN;
```

Comment:
we should add a comment to detail in which cases we would want to retry and when not.